### ISSUE TITLE
Loglikelihood in mif2 search decreases before converging

### ISSUE THREAD

**Issue by theresasophia** [kingaa/pomp#27]
I performed a global `mif2` search with the model in detailed described in Issue 25. The result I observe confuses me, because for the first few iterations the loglikelihood increases quite much, however stabilises at a  value which is below the initially found region with higher loglikelihood. In order to see the phenomenon I am talking about, the plot below does not show the first 2 iterations (the loglikelihood is starting somewhere around -10500 for the first iteration). 

![rplot01](https://cloud.githubusercontent.com/assets/8625666/19699000/cef1b146-9af1-11e6-89db-9496c13afc40.jpeg)

I tried different constellations of intensities of the random walk, used higher and lower values for the cooling fraction and also different number of particles and iterations,  but in all cases I sooner or later observe that the loglikelihood stabilizes at a value which does not look optimal after seeing higher likelihoods. Do you have any explanation for this or am I doing something wrong? Thanks a lot.

Here the code for the mif2 search:

```
sir_box <- rbind(
  beta1=c(10,13),
  beta2=c(0.2,0.6),
  beta3=c(0.2,0.6),
  beta11=c(0.11,0.15),
  phi=c(0,0.05),
  od=c(0.001,1)
)

sir_fixed_params <- c(gamma=1, delta1=1/(5*52), delta2=1/(55*52), alpha=1/(80*52), 
                      omega=1/(1*52), mu=1/(20*52), N=80000000, y1=dat$cases1[2], 
                      y2=dat$cases2[2], y3=dat$cases3[2])

stew(file="bi_box_eval_negbin-%d.rda",{

  t_global <- system.time({
    mifs_global <- foreach(i=1:20,.packages='pomp', .combine=c, .options.multicore=mcopts) %dopar% {
      mif2(
        sir,
        start=c(apply(sir_box,1,function(x)runif(1,min=x[1],max=x[2])),sir_fixed_params),
        Np=4000,
        Nmif=200,
        cooling.type="geometric",
        cooling.fraction.50=0.5,
        transform=TRUE,
        rw.sd=rw.sd(beta1=0.002,beta2=0.002,beta3=0.002,beta11=0.002,phi=0.002,od=0.01)
      )
    }
  })
},seed=1270401374,kind="L'Ecuyer")

mifs_global %>%
  conv.rec(c("loglik","nfail","beta1","beta2","beta3","beta11","phi","od")) %>%
  melt() %>%subset(iteration>2)%>%
  ggplot(aes(x=iteration,y=value,color=variable,group=L1))+
  geom_line()+
  guides(color=FALSE)+
  labs(x="MIF2 Iteration",y="")+
  facet_wrap(~variable,scales="free_y",ncol=2)+
  theme_bw()
```

**Comment by kingaa** [comment]
Thank you, @theresasophia, for bringing up this interesting issue.  My collaborators and I have observed this phenomenon on numerous occasions and believe we have something of an understanding of it.  Thanks to your clear and explicit demonstration of the issue, I will be able to rapidly turn this into a new FAQ, which should be of some help to others.

First, a question.  As you know, the `mif` likelihood is not identical to the true likelihood.  [In brief, iterated filtering turns model parameters (constants) into random variables.  As `mif` iterations proceed, the intensity of the random walks they undergo decreases.  Thus the `mif` likelihood approaches the true likelihood, but after finitely many iterations, it is not identical.  In effect, the `mif` likelihood is the likelihood of a model different from, albeit closely related to, the model of interest.  See [FAQ 5.1](http://kingaa.github.io/pomp/FAQ.html#mif-vs-pfilter-likelihood).]  Your plots above show that, after a few iterations, the `mif` likelihood displays a slow downward trend.  The question I have for you is whether the true likelihood (as estimated by `pfilter` for instance) is similarly lower at the end of the 200 `mif` iterations than it is after, say, iteration 20.

The pattern you see suggests that the `mif` model (with extra variability added via random-walk perturbations to the parameters, hereafter MM) fits the data somewhat better with more variability than it does with less.  This suggests that the model of interest (with constant parameters, hereafter MOI) may lack the variability needed to explain the data well.  In other words, a model with more variability than is possible in the MOI may be needed to explain the variability in the data, at least at the parameters you have so far explored.

Does this make sense to you?

To return to the question above: if the MOI likelihood (i.e., that returned by `pfilter`) is stable or growing, then `mif` is doing its job, which is to maximize the likelihood of the MOI (regardless of whether the MOI is the best model or even adequate).  On the other hand, if the MOI likelihood is dropping, this suggests that the `mif` cooling is happening too quickly.  Can you see why this is so?

**Comment by theresasophia** [comment]
Thanks a lot for this detailed and interesting answer. 

I was guessing my problem might be related to the fact that the likelihood of the MM is different from them MOI, so it was very helpful to get your confirmation on this. I did as you suggested and evaluated `pfilter` at 20 iterations and you were indeed right- the logliklihood returned by `pfilter`  is way lower (approx -10350 units) than the final  `pfilter` output after 100 iterations. Great!

What I find also very interesting is the other thing you mentioned- that this drop of the loglikelihood suggests that there is not enough variability in the model. I can also confirm this finding because I analysed the model with additional variability in the transmission rates. In that case, I did not observe this downward trend, however it is only now that I can make the connection that these two things are related. Thanks for the new insights!
